cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -3948,11 +4382,13 @@ macro(build_grpc)
            GRPC_C_FLAGS
            " -Wno-attributes"
            " -Wno-format-security"
+           " -Wno-return-type"
```

Comment:
I had to add those to avoid a bunch of those:
```
/build/cpp/grpc_ep-prefix/src/grpc_ep/src/core/lib/promise/party.h: In member function 'grpc_core::Poll<typename grpc_core::promise_detail::OncePromiseFactory<void, F>::Promise::Result> grpc_core::Party::PromiseParticipantImpl<SuppliedFactory>::PollCompletion() [with SuppliedFactory = grpc_core::ForwardCall(grpc_core::CallHandler, grpc_core::CallInitiator, absl::lts_20240722::AnyInvocable<void(grpc_metadata_batch&)>)::<lambda()> mutable::<lambda(std::optional<std::unique_ptr<grpc_metadata_batch, grpc_core::Arena::PooledDeleter> >)> mutable::<lambda()> mutable::<lambda(grpc_core::MessageHandle)> mutable::<lambda()>]':
/build/cpp/grpc_ep-prefix/src/grpc_ep/src/core/lib/promise/party.h:284:5: warning: control reaches end of non-void function [-Wreturn-type]
  284 |     }
      |     ^
```